runelite-client/src/main/java/net/runelite/client/plugins/config/PluginListItem.java

```diff
@@ -0,0 +1,300 @@
+	{
+		this(configPanel, null, config, configDescriptor, name, description, tags);
+	}
+
+	private PluginListItem(ConfigPanel configPanel, @Nullable Plugin plugin, @Nullable Config config,
+		@Nullable ConfigDescriptor configDescriptor, String name, String description, String... tags)
+	{
+		this.configPanel = configPanel;
+		this.plugin = plugin;
+		this.config = config;
+		this.configDescriptor = configDescriptor;
+		this.name = name;
+		this.description = description;
+		Collections.addAll(keywords, name.toLowerCase().split(" "));
+		Collections.addAll(keywords, tags);
+
+		setLayout(new BorderLayout(3, 0));
+		setPreferredSize(new Dimension(PluginPanel.PANEL_WIDTH, 20));
+
+		JLabel nameLabel = new JLabel(name);
+		nameLabel.setForeground(Color.WHITE);
+
+		if (!description.isEmpty())
+		{
+			nameLabel.setToolTipText("<html>" + name + ":<br>" + description + "</html>");
+		}
+
+		add(nameLabel, BorderLayout.CENTER);
+
+		pinButton.setPreferredSize(new Dimension(21, 0));
+		attachPinButtonListener();
+		add(pinButton, BorderLayout.LINE_START);
+
+		final JPanel buttonPanel = new JPanel();
+		buttonPanel.setLayout(new GridLayout(1, 2));
+		add(buttonPanel, BorderLayout.LINE_END);
+
+		configButton.setPreferredSize(new Dimension(25, 0));
+		configButton.setVisible(false);
+		attachConfigButtonListener();
+		buttonPanel.add(configButton);
+
+		toggleButton.setPreferredSize(new Dimension(25, 0));
+		attachToggleButtonListener(toggleButton);
+		buttonPanel.add(toggleButton);
+	}
+
+	private void attachPinButtonListener()
+	{
+		pinButton.addMouseListener(new MouseAdapter()
+		{
+			@Override
+			public void mousePressed(MouseEvent mouseEvent)
+			{
+				setPinned(!isPinned);
+				configPanel.savePinnedPlugins();
+				configPanel.openConfigList();
+			}
+		});
+	}
+
+	private void attachConfigButtonListener()
+	{
+		// no need for a listener if there are no config item to show
+		if (config == null || configDescriptor.getItems().stream().allMatch(item -> item.getItem().hidden()))
+		{
+			return;
+		}
+
+		configButton.addMouseListener(new MouseAdapter()
+		{
+			@Override
+			public void mousePressed(MouseEvent mouseEvent)
+			{
+				configButton.setIcon(CONFIG_ICON);
+				configPanel.openGroupConfigPanel(PluginListItem.this, config, configDescriptor);
+			}
+
+			@Override
+			public void mouseEntered(MouseEvent e)
+			{
+				configButton.setIcon(CONFIG_ICON_HOVER);
+			}
+
+			@Override
+			public void mouseExited(MouseEvent e)
+			{
+				configButton.setIcon(CONFIG_ICON);
+			}
+		});
+
+		configButton.setVisible(true);
+		configButton.setToolTipText("Edit plugin configuration");
+	}
+
+	private void attachToggleButtonListener(JLabel button)
+	{
+		// no need for a listener if there is no plugin to enable / disable
+		if (plugin == null)
+		{
+			button.setVisible(false);
+			return;
+		}
+
+		button.addMouseListener(new MouseAdapter()
+		{
+			@Override
+			public void mousePressed(MouseEvent mouseEvent)
```

Comment:
Prefer ActionListener to MouseAdapter as it works with keyboards. If you must use a MouseAdapter use mouseClicked rather than mousePressed